reactor-core/src/main/java/reactor/core/publisher/UnicastManySinkNoBackpressure.java

```diff
@@ -67,6 +70,18 @@
 	public void subscribe(CoreSubscriber<? super T> actual) {
 		Objects.requireNonNull(actual, "subscribe");
 
+		//detect pre-subscribe termination
+		if (STATE.compareAndSet(this, State.TERMINATED_BEFORE_SUBSCRIPTION, State.TERMINATED)) {
```

Comment:
to avoid double CASing, we can/should replace it with `do { read + switch + cas } while (casFails)`

Reply:
done, let me know if that's what you had in mind